tests/ui/iter_kv_map.rs

```diff
@@ -0,0 +1,29 @@
+// run-rustfix
+
+#![warn(clippy::iter_kv_map)]
+#![allow(clippy::redundant_clone)]
+
+use std::collections::{BTreeMap, HashMap};
+
+fn main() {
+    let map: HashMap<u32, u32> = HashMap::new();
+
+    let _ = map.iter().map(|(key, _)| key).collect::<Vec<_>>();
+    let _ = map.iter().map(|(_, value)| value).collect::<Vec<_>>();
+    let _ = map.iter().map(|(_, v)| v + 2).collect::<Vec<_>>();
+
+    let map: BTreeMap<u32, u32> = BTreeMap::new();
+
+    let _ = map.iter().map(|(key, _)| key).collect::<Vec<_>>();
+    let _ = map.iter().map(|(_, value)| value).collect::<Vec<_>>();
+    let _ = map.iter().map(|(_, v)| v + 2).collect::<Vec<_>>();
+
+    let _ = map.clone().into_iter().map(|(key, _)| key).collect::<Vec<_>>();
+    let _ = map.clone().into_iter().map(|(key, _)| key + 2).collect::<Vec<_>>();
+
+    // it seems that "method" lints simply don't trigger on this
```

Comment:
Yes, I undestand that - which is why I was surprised that this _only_ checks method calls instead of the desugared syntax. It wasn't really my intention to test against this behaviour; I'm removing it now.

Reply:
The fact that its `Expr::MethodCall` is basically the desugared syntax. There is a PR in rustc to have the self argument as an extra member in the enum. But I'm not sure what the status if that one is :)